### ISSUE TITLE
Stop working after a few hours / days of usage due to "invalid memory address or nil pointer dereference" error

### ISSUE THREAD

**Issue by pizzqc** [containous/traefik#1344]
### What version of Traefik are you using (`traefik version`)?
docker image traefik:v1.2.0 

### What is your environment & configuration (arguments, toml...)?

Using Traefik as Ingress Controllers in a Kubernetes cluster.

```
#kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:57:25Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```
```Containers:
  traefik-ingress-lb:
    Container ID:       docker://49814fb1727acb7f690baa1ba037f10cd51d2f65aa7587233a5aa770d3e8376a
    Image:              traefik:v1.2.0
    Image ID:           docker-pullable://docker.io/traefik@sha256:d9d82c52bb091466b167ea1c0f2a27c0032baef786ead275d3c40fb9e4759aaa
    Ports:              80/TCP, 8081/TCP
    Args:
      -d
      --web
      --web.address=:8081
      --kubernetes
    Requests:
      cpu:              100m
      memory:           50Mi
    State:              Running
      Started:          Fri, 24 Mar 2017 10:29:24 +0000
    Ready:              True
    Restart Count:      0
    Volume Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-92fgv (ro)
    Environment Variables:      <none>
```

### What did you do?
At first, everything works fine, all Ingresses are properly routed as expected... However, after a few hours or days depending on the usage I would guess, the ingress controllers start failing with the same `Go routine: runtime error: invalid memory address or nil pointer dereference` error

### What did you expect to see?
The server to remain stable over a long period of time without needing to be restarted

### What did you see instead?
Service is crashing with following logs:
```
time="2017-03-26T19:04:36Z" level=debug msg="Round trip: http://10.21.244.45:8081, code: 200, duration: 2.868652ms"
time="2017-03-26T19:04:37Z" level=debug msg="Round trip: http://172.17.105.4:8091, code: 200, duration: 10.018064312s"
time="2017-03-26T19:04:38Z" level=debug msg="Using label selector: ''"
time="2017-03-26T19:04:38Z" level=debug msg="Received event from kubernetes &Endpoints{ObjectMeta:ObjectMeta{Name:news,GenerateName:,Namespace:dcs-monitoring-poc,SelfLink:/api/v1/namespaces/dcs-monitori                                                                                                                                                                                                                                 ng-poc/endpoints/news,UID:74192873-04e6-11e7-bd05-000c29f3f854,ResourceVersion:4958812,Generation:0,CreationTimestamp:2017-03-09 16:35:49 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil                                                                                                                                                                                                                                 ,Labels:map[string]string{kubernetes.io/cluster-service: true,kubernetes.io/name: news,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Subsets:[{[{172.17.70.14  0xc422e                                                                                                                                                                                                                                 712f0 ObjectReference{Kind:Pod,Namespace:dcs-monitoring-poc,Name:newsrc-0c8906d06e8708c67315cbb401cc0ed8-32p8z,UID:84f5a3ff-0e7c-11e7-9a24-000c29f3f854,APIVersion:,ResourceVersion:4958810,FieldPath:,}}]                                                                                                                                                                                                                                  [] [{port 9080 TCP}]}],}"
time="2017-03-26T19:04:38Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
goroutine 38 [running]:
runtime/debug.Stack(0x4e82af, 0xc420012280, 0x1d9e5af)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1ade080, 0xc420014030)
        /go/src/github.com/containous/traefik/safe/routine.go:147 +0x7d
github.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc4229bfe68)
        /go/src/github.com/containous/traefik/safe/routine.go:155 +0x56
panic(0x1ade080, 0xc420014030)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/containous/traefik/provider.(*Kubernetes).loadIngresses(0xc420259cc0, 0x2a86780, 0xc4202a1180, 0x1, 0x1, 0x0)
        /go/src/github.com/containous/traefik/provider/kubernetes.go:169 +0x13c7
github.com/containous/traefik/provider.(*Kubernetes).Provide.func1.1(0x0, 0x0)
        /go/src/github.com/containous/traefik/provider/kubernetes.go:64 +0x324
github.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)
        /go/src/github.com/containous/traefik/safe/routine.go:159 +0x75
github.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc4204104f0, 0x2a79a80, 0xc420410500, 0x1e9b508, 0x0, 0x0)
        /go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:32 +0x3f
github.com/containous/traefik/provider.(*Kubernetes).Provide.func1(0xc420278d20)
        /go/src/github.com/containous/traefik/provider/kubernetes.go:85 +0x17f
github.com/containous/traefik/safe.(*Pool).Go.func1()
        /go/src/github.com/containous/traefik/safe/routine.go:77 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x1e9b590, 0xc4203cf440)
        /go/src/github.com/containous/traefik/safe/routine.go:141 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
        /go/src/github.com/containous/traefik/safe/routine.go:142 +0x49
time="2017-03-26T19:04:38Z" level=error msg="Kubernetes connection error Panic in operation: %!s(<nil>), retrying in 15.38715763s"
time="2017-03-26T19:04:38Z" level=debug msg="Round trip: http://10.21.244.46:8081, code: 200, duration: 1.36559ms"
```

**Current workaround:** Rollback to v1.1.2 or restart the v1.2.0 every now and then

**Comment by timoreimann** [comment]
This is a known issue that was only discovered after we released 1.2.0, unfortunately.

[The fix](https://github.com/containous/traefik/pull/1331) is already merged and will be released on Monday as 1.2.1. Apologies for the inconveniences!

**Comment by timoreimann** [comment]
I should probably summarize quickly what the issue is: there's a bug which triggers the nil dereference whenever a service referenced by an Ingress object is expected but missing.
